### DISCUSSION TITLE
Because isar_generator >=3.0.0-dev.2 <3.0.0-dev.11 depends on analyzer

### DISCUSSION THREAD

**Discussion by dsxksss** [isar/isar#1626] [Help]
flutter pub add -d isar_generator
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
The current Dart SDK version is 3.4.3.

Because isar_generator >=3.0.0-dev.2 <3.0.0-dev.11 depends on analyzer ^4.1.0 and isar_generator
  >=3.0.0-dev.0 <3.0.0-dev.2 depends on analyzer ^4.0.0, isar_generator >=3.0.0-dev.0 <3.0.0-dev.11 requires
  analyzer ^4.0.0.
And because isar_generator <0.2.0 doesn't support null safety, isar_generator <0.2.0-∞ or >=3.0.0-dev.0     
  <3.0.0-dev.11 requires analyzer ^4.0.0.
And because isar_generator >=0.2.0 <1.0.0 depends on freezed_annotation ^0.14.1 and isar_generator >=1.0.0  
  <2.0.0-dev.0 depends on analyzer ^2.0.0, isar_generator <2.0.0-dev.0 or >=3.0.0-dev.0 <3.0.0-dev.11       
  requires analyzer ^2.0.0 or ^4.0.0 or freezed_annotation ^0.14.1.
And because isar_generator >=2.0.0-dev.0 <3.0.0-dev.0 depends on analyzer ^3.0.0 and isar_generator
  >=3.0.0-dev.11 <3.0.1 depends on analyzer ^4.6.0, isar_generator <3.0.1 requires analyzer ^2.0.0 or ^3.0.0
  or >=4.0.0 <5.0.0 or freezed_annotation ^0.14.1.
And because isar_generator >=3.0.1 depends on analyzer >=4.6.0 <6.0.0 and flutter_desktop_os depends on     
  freezed_annotation ^2.4.1, every version of isar_generator requires analyzer ^2.0.0 or ^3.0.0 or >=4.0.0  
  <6.0.0.
So, because flutter_desktop_os depends on both analyzer ^6.3.0 and isar_generator any, version solving      
  failed.

The lower bound of "sdk: '>=2.8.1 <3.0.0'" must be 2.12.0 or higher to enable null safety.
For details, see https://dart.dev/null-safety

how can fix this. 🤔

**Comment by al-mohad** [comment]
@dsxksss 
Use   wolt_modal_sheet: 0.8.0
